fix: Improve caching mechanism for URL indexing status

pmespresso opened the pull request:
**What did I change?**

- Modify shouldRecheck function to always process non-indexed URLs
- Ensure "Submitted and Indexed" URLs are only rechecked after cache timeout
- Optimize batch processing to avoid unnecessary API calls

**Why did I change it?**

Every run on the script was trying the process the same urls and then timing out. 
Related: https://github.com/goenning/google-indexing-script/issues/51 https://github.com/goenning/google-indexing-script/pull/59

pmespresso committed:
fix: Improve caching mechanism for URL indexing status

- Modify shouldRecheck function to always process non-indexed URLs
- Ensure "Submitted and Indexed" URLs are only rechecked after cache timeout
- Optimize batch processing to avoid unnecessary API calls

goenning commented on src/index.ts:
```diff
@@ -120,11 +120,20 @@ export const index = async (input: string = process.argv[2], options: IndexOptio
 
   const shouldRecheck = (status: Status, lastCheckedAt: string) => {
-    const shouldIndexIt = indexableStatuses.includes(status);
+    if (status !== Status.SubmittedAndIndexed) {
```
this seems wrong, there are some statuses that should not force a recheck

or am I missing something?

pmespresso replied:
Ah you're right, I overfit to my own personal issue.

My cache was saving "status": "Not found (404)", which is not currently part of the Status enum, so those were being skipped.

I should rather just add a new Status.NotFound to indexableStatus.

However, such things can be fixed quite quickly but because of the long cacheTimeout they'd be ignored for 2 weeks. Perhaps we can have a shorter timeout for NotFound or PageWithRedirect. 

What do you think?

pmespresso committed:
- Introduce short-term cache for quickly fixable statuses (e.g., NotFound)
- Maintain efficient long-term caching for stable, indexed URLs
- add NotFound to Status enum
